apps/bfd-pipeline/bfd-pipeline-rif-extract/src/test/java/gov/cms/bfd/pipeline/rif/extract/synthetic/SyntheticDataFixer4.java

```diff
@@ -0,0 +1,308 @@
+package gov.cms.bfd.pipeline.rif.extract.synthetic;
+
+import gov.cms.bfd.model.rif.BeneficiaryColumn;
+import gov.cms.bfd.model.rif.RifFileType;
+import gov.cms.bfd.model.rif.parse.RifParsingUtils;
+import gov.cms.bfd.pipeline.rif.extract.LocalRifFile;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVParser;
+import org.apache.commons.csv.CSVPrinter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A one-off utility application for modifying the fixed-with-negative-ids sample data that was
+ * produced on 2017-11-27 and then fixed by {@link SyntheticDataFixer}. This class adds additional
+ * beneficiary columns from the BSF file.
+ */
+public final class SyntheticDataFixer4 {
+  private static class MBIPool {
```

Comment:
I'd like to see some comments or code or something explaining whether or not these MBIs can be expected to **ever** collide with MBIs from real beneficiaries.

This PR should also include updates to `beneficiary-fhir-data/apps/bfd-model/bfd-model-rif-samples/dev/design-sample-data-sets.md` along those lines.

Reply:
Makes sense, can you share the updates you made to that document that define the format for these synthetic MBIs? I'll toss those udpates into this PR.